crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,1203 @@
+  name: String!
+  struct: MoveStructDecl!
+}
+
+# Represents concrete types (no type parameters, no references)
+type MoveType {
+  # Scalar representation of the type instantiation (type and type
+  # parameters)
+  repr: String!
+  typeName: MoveTypeName!
+  typeParameters: [MoveType]
+}
+
+# Represents types that could contain unbound type parameters, or
+# references. Such types can appear as function parameters, or fields
+# in structs.
+type OpenMoveType {
+  repr: String!
+  ref: MoveReference
+  type: OpenMoveTypeBody
+}
+
+union OpenMoveTypeBody = MoveTypeApplicatioon | MoveTypeParameter
+
+type MoveTypeApplication {
+  name: MoveTypeName!
+  typeParameters: [OpenMoveType]
+}
+
+type MoveTypeParameter {
+  index: Int!
+}
+
+# Metrics (omitted for brevity)
+type NetworkMetrics {}
+type MoveCallMetrics {}
+type AddressMetrics {
+  id: ID!
+}
+
+# Execution
+
+type ExecutionResult {
+  effectsDigest: String
+  errors: String
+}
+
+type DryRunResult {
+  transaction: TransactionBlock
+  errors: String
+
+  events: [Event!]
+  results: [DryRunEffect!]
+}
+
+type DryRunEffect {
+  # Changes made to arguments that were mutably borrowed by this
+  # transaction
+  mutatedReferences: [DryRunMutation!]
+
+  # Results of this transaction
+  returnValues: [DryRunReturn!]
+}
+
+type DryRunMutation {
+  input: TransactionInput
+  type: MoveType
+  bcs: Base64
+}
+
+type DryRunReturn {
+  type: MoveType
+  bcs: Base64
+}
+
+# Connections
+
+# Pagination
+type PageInfo {
+    hasNextPage: Boolean!
+    hasPreviousPage: Boolean!
+    startCursor: String
+    endCursor: String
+}
+
+# Checkpoints
+type CheckpointConnection {
+  edges: [CheckpointEdge!]!
+  pageInfo: PageInfo!
+}
+
+type CheckpointEdge {
+  cursor: String
+  node: Checkpoint!
+}
+
+# Balance
+type BalanceConnection {
+  edges: [BalanceEdge!]!
+  pageInfo: PageInfo!
+}
+
+type BalanceEdge {
+  cursor: String
+  node: Balance!
+}
+
+# Coin
+type CoinConnection {
+  edges: [CoinEdge!]!
+  pageInfo: PageInfo!
+}
+
+type CoinEdge {
+  cursor: String
+  node: Coin!
+}
+
+# DynamicField
+type DynamicFieldConnection {
+  edges: [DynamicFieldEdge!]!
+  pageInfo: PageInfo!
+}
+
+type DynamicFieldEdge {
+  cursor: String
+  node: DynamicField!
+}
+
+# Object
+type ObjectConnection {
+  edges: [ObjectEdge!]!
+  pageInfo: PageInfo!
+}
+
+type ObjectEdge {
+  cursor: String
+  node: Object!
+}
+
+# Event
+type EventConnection {
+  edges: [EventEdge!]!
+  pageInfo: PageInfo!
+}
+
+type EventEdge {
+  cursor: String
+  node: Event!
+}
+
+# MoveFunction
+type MoveFunctionConnection {
+  edges: [MoveFunctionEdge!]!
+  pageInfo: PageInfo!
+}
+
+type MoveFunctionEdge {
+  cursor: String
+  node: MoveFunction!
+}
+
+# MoveStructDeclConnection
+type MoveStructConnection {
+  edges: [MoveStructEdge!]!
+  pageInfo: PageInfo!
+}
+
+type MoveStructEdge {
+  cursor: String
+  node: MoveStructDecl!
+}
+
+# TransactionBlockConnection
+type TransactionBlockConnection {
+  totalTransactionBlocks: Int
```

Comment:
Doing some back of the envelope maths (to check on `Int` vs `BigInt` here), if we are sustaining 1K PTB/s, this gives us runway for almost 300,000 years, so even if we're sustaining 1M PTB/s, we're okay for 300 years.

Technically speaking I think you can make the same argument for object versions and checkpoint sequence numbers, since those can only be as big as this number.  Thoughts @Jordan-Mysten, @hayes-mysten should those also be `Int`s?

Reply:
I’m happy with that. Changing these to bigints if they get too big isn’t that hard anyway, and leaving them as numbers make developer lives a little nicer in the interim.